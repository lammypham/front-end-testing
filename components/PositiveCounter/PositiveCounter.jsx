import React, { Component } from 'react';

export default class PositiveCounter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };

    this.increment = this.increment.bind(this);
    this.decrement = this.decrement.bind(this);
  }

  increment() {
    this.setState({
      count: this.state.count + 1
    });
  }

  decrement(){
    this.setState({
      count: Math.max(0, this.state.count - 1)
    });
  }

  render() {
    return (
      <div>
        Value: {this.state.count}
        <button className="decrement" onClick={this.decrement}>&minus;</button>
        <button className="increment" onClick={this.increment}>+</button>
      </div>
    );
  }
}