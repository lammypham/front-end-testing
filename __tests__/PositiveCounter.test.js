import React from 'react';
import ReactDOM from 'react-dom';
import PositiveCounter from '../components/Positivecounter/PositiveCounter';
import renderer from 'react-test-renderer';

it('should render 0', () => {
    const tree = renderer.create(<PositiveCounter />).toJSON();
    expect(tree).toMatchSnapshot();
})

it('should render 2', () => {
    const component = renderer.create(<PositiveCounter />);
    component.getInstance().increment();
    component.getInstance().increment();
    expect(component.toJSON()).toMatchSnapshot();
})

it('should not go negative', () => {
    const component = renderer.create(<PositiveCounter />);
    component.getInstance().increment();
    component.getInstance().decrement();
    component.getInstance().decrement();
    expect(component.toJSON()).toMatchSnapshot();
})