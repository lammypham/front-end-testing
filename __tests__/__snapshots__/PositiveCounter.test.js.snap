// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not go negative 1`] = `
<div>
  Value: 
  0
  <button
    className="decrement"
    onClick={[Function]}
  >
    −
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`should render 0 1`] = `
<div>
  Value: 
  0
  <button
    className="decrement"
    onClick={[Function]}
  >
    −
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`should render 2 1`] = `
<div>
  Value: 
  2
  <button
    className="decrement"
    onClick={[Function]}
  >
    −
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
